test/cppsuite/tests/bounded_cursor_perf.cpp

```diff
@@ -0,0 +1,110 @@
+/*-
+ * Public Domain 2014-present MongoDB, Inc.
+ * Public Domain 2008-2014 WiredTiger, Inc.
+ *
+ * This is free and unencumbered software released into the public domain.
+ *
+ * Anyone is free to copy, modify, publish, use, compile, sell, or
+ * distribute this software, either in source code form or as a compiled
+ * binary, for any purpose, commercial or non-commercial, and by any
+ * means.
+ *
+ * In jurisdictions that recognize copyright laws, the author or authors
+ * of this software dedicate any and all copyright interest in the
+ * software to the public domain. We make this dedication for the benefit
+ * of the public at large and to the detriment of our heirs and
+ * successors. We intend this dedication to be an overt act of
+ * relinquishment in perpetuity of all present and future rights to this
+ * software under copyright law.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include <time.h>
+
+#include "test_harness/core/op_tracker.cpp"
+#include "test_harness/test.h"
+#include "test_harness/util/api_const.h"
+#include "test_harness/workload/random_generator.h"
+
+namespace test_harness {
+
+/*
+ * This test performs cursor traversal operations next() and prev() on a collection with both
+ * bounded and normal cursors. The performance of both cursors are tracked and the average time
+ * taken is added to the perf file. The test traverses all keys in the collection.
+ */
+class bounded_cursor_perf : public test {
+    public:
+    bounded_cursor_perf(const test_args &args) : test(args) {}
+
+    void
+    read_operation(thread_context *tc) override final
+    {
+        /* This test will only work with one read thread. */
+        testutil_assert(tc->thread_count == 1);
+        /*
+         * Each read operation performs next() and prev() calls with both normal cursors and bounded
+         * cursors.
+         */
+        int range_ret_next, range_ret_prev, ret_next, ret_prev;
+
+        /* Initialize the op trackers. */
+        op_tracker bounded_next("bounded_next", test::_args.test_name);
+        op_tracker default_next("default_next", test::_args.test_name);
+        op_tracker bounded_prev("bounded_prev", test::_args.test_name);
+        op_tracker default_prev("default_prev", test::_args.test_name);
+
+        /* Get the collection to work on. */
+        collection &coll = tc->db.get_collection(0);
```

Comment:
I would choose to get a random collection. 
```
collection &coll = tc->db.get_random_collection();
```

Reply:
What are your thoughts on this @luke-pearson ? I think you mentioned preferring the`get_collection()` approach.

Reply:
I think we should just assert that collection count is 1 prior to this call, and stick to getting the 1st collection.